README.md
<p align="left">
  <a href="https://github.com/bentzibentz/tailwindcss-animate.css/releases" target="_blank"><img alt="GitHub release" src="https://img.shields.io/github/v/release/bentzibentz/tailwindcss-animate.css?include_prereleases&style=flat-square"></a>
  <a href="https://npmjs.com/package/tailwindcss-animatecss" target="_blank"><img alt="npm" src="https://img.shields.io/npm/v/tailwindcss-animatecss?style=flat-square"></a>
  <a href="https://npmjs.com/package/tailwindcss-animatecss" target="_blank"><img alt="npm" src="https://img.shields.io/npm/dt/tailwindcss-animatecss?style=flat-square"></a>
  <img alt="npm bundle size" src="https://img.shields.io/bundlephobia/minzip/tailwindcss-animatecss?style=flat-square">
  <img alt="GitHub top language" src="https://img.shields.io/github/languages/top/bentzibentz/tailwindcss-animate.css?style=flat-square">
  <img alt="License" src="https://img.shields.io/github/license/bentzibentz/tailwindcss-animate.css">
</p>

# Animate CSS - Tailwind CSS Plugin
Add [Animate CSS](https://github.com/animate-css/animate.css) as [Tailwind CSS](https://github.com/tailwindcss/tailwindcss) plugin to your project.

## Important Information

Tailwindcss >= v3.1.8 is required.

## Installation

Pull it in through npm or yarn:

```bash
npm install tailwindcss-animatecss
```

```bash
yarn add tailwindcss-animatecss
```

Use animatecss < v4:

```bash
npm install tailwindcss-animatecss@0.3.3
```

```bash
yarn add tailwindcss-animatecss@0.3.3
```

### Usage with Tailwind v1 and v2

For Documentation checkout Branch ``feature/v2``

```bash
npm install tailwindcss-animatecss@1.0.8
```

```bash
yarn add tailwindcss-animatecss@1.0.8
```

### Usage with old default class names ``.animate--``

For Documentation checkout Branch ``feature/v3``

```bash
npm install tailwindcss-animatecss@3.0.2
```

```bash
yarn add tailwindcss-animatecss@3.0.2
```

## Usage

Add Tailwind CSS to your project as described [here](https://tailwindcss.com/docs/installation).
Install tailwindcss plugin as described above and add it to your tailwind.config.js file. 

```js
plugins: [
    // Other plugins
    require('tailwindcss-animatecss'),
]
```

You want to determine yourself which classes are used? You just have to set these class names at the classes array. For example if you want ```.ani-fadeIn``` just add ```fadeIn``` to the array.
You can find all available class names further down.

Defining the classes is recommended to avoid to bloat your css with unused classes and keyframes.

```js
// tailwind.config.js
module.exports = {
    theme: {
        animatedSettings: {
            defaultClassName: true,
            animatedSpeed: 1000,
            animatedSpeed: 1000,
            heartBeatSpeed: 500,
            hingeSpeed: 2000,
            bounceInSpeed: 750,
            bounceOutSpeed: 750,
            animationDelaySpeed: 500,
            classes: ['bounce', 'heartBeat']
        }
    },
}
```

If you want to prefix your CSS classes, use the tailwind prefix option:

```js
// tailwind.config.js
module.exports = {
  prefix: 'tw-',
  // ...
}
```
### Adjustable, optional settings
* ```defaultClassName``` used for the default class name generation, like ```.ani-fadeIn```, set it to ``false`` to generate classes without the default class name like like ```.fadeIn```
* ```animatedSpeed``` used for the animationDuration of the .animated class
* ```heartBeat``` used for the animationDuration of the .heartBeat class
* ```hingeSpeed``` used for the animationDuration of the .hingeSpeed class
* ```bounceInSpeed``` used for the animationDuration of the .bounceInSpeed class
* ```bounceOutSpeed``` used for the animationDuration of the .bounceOutSpeed class
* ```animationDelaySpeed``` used for the animationDelaySpeed, just add .delay class to your element

All of these settings are optional, if not set basic animate.css fallback animationDuration speed is used.

The plugin generates all the animate.css utility classes for you.

### Variants
Generating different class variants is super easy, tailwind does all the work for you.

### Available Animate CSS classes
* .ani-animated
* .ani-infinite
* .ani-delay
* .ani-delay-1s
* .ani-delay-2s
* .ani-delay-3s
* .ani-delay-4s
* .ani-delay-5s
* .ani-fast
* .ani-faster
* .ani-slow
* .ani-slower
* .ani-bounce
* .ani-flash
* .ani-pulse
* .ani-rubberBand
* .ani-shakeY
* .ani-shakeX
* .ani-headShake
* .ani-swing
* .ani-tada
* .ani-wobble
* .ani-jello
* .ani-heartBeat
* .ani-hinge
* .ani-jackInTheBox
* .ani-lightSpeedIn
* .ani-lightSpeedOut
* .ani-flip
* .ani-flipInX
* .ani-flipInY
* .ani-flipOutX
* .ani-flipOutY
* .ani-rotateIn
* .ani-rotateInDownLeft
* .ani-rotateInDownRight
* .ani-rotateInUpLeft
* .ani-rotateInUpRight
* .ani-rotateOut
* .ani-rotateOutDownLeft
* .ani-rotateOutDownRight
* .ani-rotateOutUpLeft
* .ani-rotateOutUpRight
* .ani-rollIn
* .ani-rollOut
* .ani-zoomIn
* .ani-zoomInUp
* .ani-zoomInDown
* .ani-zoomInLeft
* .ani-zoomInRight
* .ani-bounceIn
* .ani-bounceInDown
* .ani-bounceInUp
* .ani-bounceInLeft
* .ani-bounceInRight
* .ani-bounceOut
* .ani-bounceOutDown
* .ani-bounceOutUp
* .ani-bounceOutLeft
* .ani-bounceOutRight
* .ani-slideInDown
* .ani-slideInLeft
* .ani-slideInRight
* .ani-slideInUp
* .ani-slideOutDown
* .ani-slideOutLeft
* .ani-slideOutRight
* .ani-slideOutUp
* .ani-fadeIn
* .ani-fadeInDown
* .ani-fadeInDownBig
* .ani-fadeInLeft
* .ani-fadeInLeftBig
* .ani-fadeInRight
* .ani-fadeInRightBig
* .ani-fadeInUp
* .ani-fadeInUpBig
* .ani-fadeInTopLeft
* .ani-fadeInTopRight
* .ani-fadeInBottomLeft
* .ani-fadeInBottomRight
* .ani-fadeOut
* .ani-fadeOutDown
* .ani-fadeOutDownBig
* .ani-fadeOutLeft
* .ani-fadeOutLeftBig
* .ani-fadeOutRight
* .ani-fadeOutRightBig
* .ani-fadeOutUp
* .ani-fadeOutUpBig
* .ani-fadeOutTopLeft
* .ani-fadeOutTopRight
* .ani-fadeOutBottomLeft
* .ani-fadeOutBottomRight
* .ani-zoomOutDown
* .ani-zoomOutLeft
* .ani-zoomOutRight
* .ani-zoomOut
* .ani-zoomOutUp
* .ani-lightSpeedInRight
* .ani-lightSpeedInLeft
* .ani-lightSpeedOutRight
* .ani-lightSpeedOutLeft
* .ani-backInDown
* .ani-backInUp
* .ani-backInLeft
* .ani-backInRight
* .ani-backOutDown
* .ani-backOutUp
* .ani-backOutLeft
* .ani-backOutRight

## Demo

Coming soon…

## Credits

This package is based on awesome [Animate.css](https://github.com/animate-css/animate.css).

## More Tailwind CSS Plugins

[tailwindcss-layered-box-shadow](https://github.com/bentzibentz/tailwindcss-layered-box-shadow)

[tailwindcss-absolute-center](https://github.com/bentzibentz/tailwindcss-absolute-center)

## License

This project is licensed under the [MIT License](https://opensource.org/licenses/MIT).


CHANGELOG.md


## [3.0.5](https://github.com/bentzibentz/tailwindcss-animate.css/compare/v3.0.4...v3.0.5) (2023-12-02)

### [3.0.4](https://github.com/bentzibentz/tailwindcss-animate.css/compare/v3.0.3...v3.0.4) (2023-03-30)

### [3.0.3](https://github.com/bentzibentz/tailwindcss-animate.css/compare/v3.0.2...v3.0.3) (2023-03-15)


### Bug Fixes

* add missing scaleY from bounce animation ([a5ea793](https://github.com/bentzibentz/tailwindcss-animate.css/commit/a5ea793d676501c383e67c86c7ffbe50ec233d99))

### [3.0.2](https://github.com/bentzibentz/tailwindcss-animate.css/compare/v3.0.1...v3.0.2) (2022-11-21)

## [3.0.0](https://github.com/bentzibentz/tailwindcss-animate.css/compare/v1.0.8...v3.0.0) (2022-09-03)


src/index.js
const plugin = require('tailwindcss/plugin');
const keyframes = require('./keyframes/keyframes');
const keyframesLightSpeed = require('./keyframes/lightspeed');
const keyframesFlip = require('./keyframes/flip');
const keyframesRotateIn = require('./keyframes/rotateIn');
const keyframesRotateOut = require('./keyframes/rotateOut');
const keyframesRoll = require('./keyframes/roll');
const keyframesZoomIn = require('./keyframes/zoomIn');
const keyframesZoomOut = require('./keyframes/zoomOut');
const keyframesBounceIn = require('./keyframes/bounceIn');
const keyframesBounceOut = require('./keyframes/bounceOut');
const keyframesSlideIn = require('./keyframes/slideIn');
const keyframesSlideOut = require('./keyframes/slideOut');
const keyframesFadeIn = require('./keyframes/fadeIn');
const keyframesFadeOut = require('./keyframes/fadeOut');
const keyframesBackIn = require('./keyframes/backIn');
const keyframesBackOut = require('./keyframes/backOut');

const defaultAnimatedClassName = 'animate__';

const animateCssTailwindCss = plugin(
    function ({addUtilities, theme, e}) {
        let animatedSettings = theme('animatedSettings');

        if (!animatedSettings) {
            animatedSettings = {};
        }

        // set fallback if speed not defined
        const animatedSpeed = animatedSettings.animatedSpeed ? animatedSettings.animatedSpeed : 1000;
        const heartBeatSpeed = animatedSettings.heartBeatSpeed ? animatedSettings.heartBeatSpeed : 1000;
        const hingeSpeed = animatedSettings.hingeSpeed ? animatedSettings.hingeSpeed : 2000;
        const bounceInSpeed = animatedSettings.bounceInSpeed ? animatedSettings.bounceInSpeed : 750;
        const bounceOutSpeed = animatedSettings.bounceOutSpeed ? animatedSettings.bounceOutSpeed : 750;
        const animationDelaySpeed = animatedSettings.animationDelaySpeed ? animatedSettings.animationDelaySpeed : 500;
        const classes = animatedSettings.classes ? animatedSettings.classes : [];
        // perform only undefined check to allow empty string/class name
        const animatedClassName = animatedSettings.animatedClassName === undefined ? defaultAnimatedClassName : animatedSettings.animatedClassName;

        const fallbackKeyframes = {
            '@keyframes bounce': keyframes.keyframeBounce,
            '@keyframes flash': keyframes.keyframeFlash,
            '@keyframes pulse': keyframes.keyframePulse,
            '@keyframes rubberBand': keyframes.keyframeRubberBand,
            '@keyframes shakeX': keyframes.keyframeShakeX,
            '@keyframes shakeY': keyframes.keyframeShakeY,
            '@keyframes headShake': keyframes.keyframeHeadShake,
            '@keyframes swing': keyframes.keyframeSwing,
            '@keyframes tada': keyframes.keyframeTada,
            '@keyframes wobble': keyframes.keyframeWobble,
            '@keyframes jello': keyframes.keyframeJello,
            '@keyframes heartBeat': keyframes.keyframeHeartBeat,
            '@keyframes hinge': keyframes.keyframeHinge,
            '@keyframes jackInTheBox': keyframes.keyframeJackInTheBox,
            '@keyframes lightSpeedInLeft': keyframesLightSpeed.keyframeLightSpeedInLeft,
            '@keyframes lightSpeedInRight': keyframesLightSpeed.keyframeLightSpeedInRight,
            '@keyframes lightSpeedOutLeft': keyframesLightSpeed.keyframeLightSpeedOutLeft,
            '@keyframes lightSpeedOutRight': keyframesLightSpeed.keyframeLightSpeedOutRight,
            '@keyframes flip': keyframesFlip.keyframeFlip,
            '@keyframes flipInX': keyframesFlip.keyframeFlipInX,
            '@keyframes flipInY': keyframesFlip.keyframeFlipInY,
            '@keyframes flipOutX': keyframesFlip.keyframeFlipOutX,
            '@keyframes flipOutY': keyframesFlip.keyframeFlipOutY,
            '@keyframes rotateIn': keyframesRotateIn.keyframeRotateIn,
            '@keyframes rotateInDownLeft': keyframesRotateIn.keyframeRotateInDownLeft,
            '@keyframes rotateInDownRight': keyframesRotateIn.keyframeRotateInDownRight,
            '@keyframes rotateInUpLeft': keyframesRotateIn.keyframeRotateInUpLeft,
            '@keyframes rotateInUpRight': keyframesRotateIn.keyframeRotateInUpRight,
            '@keyframes rotateOut': keyframesRotateOut.keyframeRotateOut,
            '@keyframes rotateOutDownLeft': keyframesRotateOut.keyframeRotateOutDownLeft,
            '@keyframes rotateOutDownRight': keyframesRotateOut.keyframeRotateOutDownRight,
            '@keyframes rotateOutUpLeft': keyframesRotateOut.keyframeRotateOutUpLeft,
            '@keyframes rotateOutUpRight': keyframesRotateOut.keyframeRotateOutUpRight,
            '@keyframes rollIn': keyframesRoll.keyframeRollIn,
            '@keyframes rollOut': keyframesRoll.keyframeRollOut,
            '@keyframes zoomIn': keyframesZoomIn.keyframeZoomIn,
            '@keyframes zoomInDown': keyframesZoomIn.keyframeZoomInDown,
            '@keyframes zoomInLeft': keyframesZoomIn.keyframeZoomInLeft,
            '@keyframes zoomInRight': keyframesZoomIn.keyframeZoomInRight,
            '@keyframes zoomInUp': keyframesZoomIn.keyframeZoomInUp,
            '@keyframes bounceIn': keyframesBounceIn.keyframeBounceIn,
            '@keyframes bounceInDown': keyframesBounceIn.keyframeBounceInDown,
            '@keyframes bounceInLeft': keyframesBounceIn.keyframeBounceInLeft,
            '@keyframes bounceInRight': keyframesBounceIn.keyframeBounceInRight,
            '@keyframes bounceInUp': keyframesBounceIn.keyframeBounceInUp,
            '@keyframes bounceOut': keyframesBounceOut.keyframeBounceOut,
            '@keyframes bounceOutDown': keyframesBounceOut.keyframeBounceOutDown,
            '@keyframes bounceOutLeft': keyframesBounceOut.keyframeBounceOutLeft,
            '@keyframes bounceOutRight': keyframesBounceOut.keyframeBounceOutRight,
            '@keyframes bounceOutUp': keyframesBounceOut.keyframeBounceOutUp,
            '@keyframes zoomOut': keyframesZoomOut.keyframeZoomOut,
            '@keyframes zoomOutDown': keyframesZoomOut.keyframeZoomOutDown,
            '@keyframes zoomOutLeft': keyframesZoomOut.keyframeZoomOutLeft,
            '@keyframes zoomOutRight': keyframesZoomOut.keyframeZoomOutRight,
            '@keyframes zoomOutUp': keyframesZoomOut.keyframeZoomOutUp,
            '@keyframes slideInDown': keyframesSlideIn.keyframeSlideInDown,
            '@keyframes slideInLeft': keyframesSlideIn.keyframeSlideInLeft,
            '@keyframes slideInRight': keyframesSlideIn.keyframeSlideInRight,
            '@keyframes slideInUp': keyframesSlideIn.keyframeSlideInUp,
            '@keyframes slideOutDown': keyframesSlideOut.keyframeSlideOutDown,
            '@keyframes slideOutLeft': keyframesSlideOut.keyframeSlideOutLeft,
            '@keyframes slideOutRight': keyframesSlideOut.keyframeSlideOutRight,
            '@keyframes slideOutUp': keyframesSlideOut.keyframeSlideOutUp,
            '@keyframes fadeIn': keyframesFadeIn.keyframeFadeIn,
            '@keyframes fadeInDown': keyframesFadeIn.keyframeFadeInDown,
            '@keyframes fadeInDownBig': keyframesFadeIn.keyframeFadeInDownBig,
            '@keyframes fadeInLeft': keyframesFadeIn.keyframeFadeInLeft,
            '@keyframes fadeInLeftBig': keyframesFadeIn.keyframeFadeInLeftBig,
            '@keyframes fadeInRight': keyframesFadeIn.keyframeFadeInRight,
            '@keyframes fadeInRightBig': keyframesFadeIn.keyframeFadeInRightBig,
            '@keyframes fadeInTopLeft': keyframesFadeIn.keyframeFadeInTopLeft,
            '@keyframes fadeInTopRight': keyframesFadeIn.keyframeFadeInTopRight,
            '@keyframes fadeInBottomLeft': keyframesFadeIn.keyframeFadeInBottomLeft,
            '@keyframes fadeInBottomRight': keyframesFadeIn.keyframeFadeInBottomRight,
            '@keyframes fadeInUp': keyframesFadeIn.keyframeFadeInUp,
            '@keyframes fadeInUpBig': keyframesFadeIn.keyframeFadeInUpBig,
            '@keyframes fadeOut': keyframesFadeOut.keyframeFadeOut,
            '@keyframes fadeOutDown': keyframesFadeOut.keyframeFadeOutDown,
            '@keyframes fadeOutDownBig': keyframesFadeOut.keyframeFadeOutDownBig,
            '@keyframes fadeOutLeft': keyframesFadeOut.keyframeFadeOutLeft,
            '@keyframes fadeOutLeftBig': keyframesFadeOut.keyframeFadeOutLeftBig,
            '@keyframes fadeOutRight': keyframesFadeOut.keyframeFadeOutRight,
            '@keyframes fadeOutRightBig': keyframesFadeOut.keyframeFadeOutRightBig,
            '@keyframes fadeOutUp': keyframesFadeOut.keyframeFadeOutUp,
            '@keyframes fadeOutUpBig': keyframesFadeOut.keyframeFadeOutUpBig,
            '@keyframes fadeOutTopLeft': keyframesFadeOut.keyframeFadeOutTopLeft,
            '@keyframes fadeOutTopRight': keyframesFadeOut.keyframeFadeOutTopRight,
            '@keyframes fadeOutBottomLeft': keyframesFadeOut.keyframeFadeOutBottomLeft,
            '@keyframes fadeOutBottomRight': keyframesFadeOut.keyframeFadeOutBottomRight,
            '@keyframes backInDown': keyframesBackIn.keyframeBackInDown,
            '@keyframes backInUp': keyframesBackIn.keyframeBackInUp,
            '@keyframes backInLeft': keyframesBackIn.keyframeBackInLeft,
            '@keyframes backInRight': keyframesBackIn.keyframeBackInRight,
            '@keyframes backOutDown': keyframesBackOut.keyframeBackOutDown,
            '@keyframes backOutUp': keyframesBackOut.keyframeBackOutUp,
            '@keyframes backOutLeft': keyframesBackOut.keyframeBackOutLeft,
            '@keyframes backOutRight': keyframesBackOut.keyframeBackOutRight,
        }

        const fallbackAnimateUtility = {
            'animated': {
                'animation-duration': `${animatedSpeed}ms`,
                'animation-fill-mode': 'both'
            },
        }

        const fallbackUtilities = {
            'infinite': {
                'animation-iteration-count': 'infinite'
            },
            'repeat-1': {
                'animation-repeat': 1
            },
            'repeat-2': {
                'animation-repeat': 2
            },
            'repeat-3': {
                'animation-repeat': 3
            },
            'delay': {
                'animation-delay': `${animationDelaySpeed}ms`
            },
            'delay-1s': {
                'animation-delay': `1000ms`
            },
            'delay-2s': {
                'animation-delay': `2000ms`
            },
            'delay-3s': {
                'animation-delay': `3000ms`
            },
            'delay-4s': {
                'animation-delay': `4000ms`
            },
            'delay-5s': {
                'animation-delay': `5000ms`
            },
            'fast': {
                'animation-duration': `800ms`
            },
            'faster': {
                'animation-duration': `500ms`
            },
            'slow': {
                'animation-duration': `2000ms`
            },
            'slower': {
                'animation-duration': `3000ms`
            },
            'bounce': {
                'animation-name': 'bounce',
                'transform-origin': 'center bottom'
            },
            'flash': {
                'animation-name': 'flash',
            },
            'pulse': {
                'animation-name': 'pulse',
            },
            'rubberBand': {
                'animation-name': 'rubberBand',
            },
            'shakeX': {
                'animation-name': 'shakeX',
            },
            'shakeY': {
                'animation-name': 'shakeY',
            },
            'headShake': {
                'animation-timing-function': 'ease-in-out',
                'animation-name': 'headShake',
            },
            'swing': {
                'transform-origin': 'top center',
                'animation-name': 'swing',
            },
            'tada': {
                'animation-name': 'tada',
            },
            'wobble': {
                'animation-name': 'wobble',
            },
            'jello': {
                'animation-name': 'jello',
            },
            'heartBeat': {
                'animation-name': 'heartBeat',
                'animation-duration': `${heartBeatSpeed}ms`,
                'animation-timing-function': 'ease-in-out'
            },
            'hinge': {
                'animation-name': 'hinge',
                'animation-duration': `${hingeSpeed}ms`,
            },
            'jackInTheBox': {
                'animation-name': 'jackInTheBox',
            },
            'lightSpeedInLeft': {
                'animation-name': 'lightSpeedInLeft',
            },
            'lightSpeedInRight': {
                'animation-name': 'lightSpeedInRight',
            },
            'lightSpeedOutLeft': {
                'animation-name': 'lightSpeedOutLeft',
            },
            'lightSpeedOutRight': {
                'animation-name': 'lightSpeedOutRight',
            },
            'flip': {
                'animation-name': 'flip',
                'backface-visibility': 'visible'
            },
            'flipInX': {
                'animation-name': 'flipInX',
                'backface-visibility': 'visible'
            },
            'flipInY': {
                'animation-name': 'flipInY',
                'backface-visibility': 'visible'
            },
            'flipOutX': {
                'animation-name': 'flipOutX',
                'backface-visibility': 'visible'
            },
            'flipOutY': {
                'animation-name': 'flipOutY',
                'backface-visibility': 'visible'
            },
            'rotateIn': {
                'animation-name': 'rotateIn',
            },
            'rotateInDownLeft': {
                'animation-name': 'rotateInDownLeft',
            },
            'rotateInDownRight': {
                'animation-name': 'rotateInDownRight',
            },
            'rotateInUpLeft': {
                'animation-name': 'rotateInUpLeft',
            },
            'rotateInUpRight': {
                'animation-name': 'rotateInUpRight',
            },
            'rotateOut': {
                'animation-name': 'rotateOut',
            },
            'rotateOutDownLeft': {
                'animation-name': 'rotateOutDownLeft',
            },
            'rotateOutDownRight': {
                'animation-name': 'rotateOutDownRight',
            },
            'rotateOutUpLeft': {
                'animation-name': 'rotateOutUpLeft',
            },
            'rotateOutUpRight': {
                'animation-name': 'rotateOutUpRight',
            },
            'rollIn': {
                'animation-name': 'rollIn',
            },
            'rollOut': {
                'animation-name': 'rollOut',
            },
            'zoomIn': {
                'animation-name': 'zoomIn',
            },
            'zoomInDown': {
                'animation-name': 'zoomInDown',
            },
            'zoomInLeft': {
                'animation-name': 'zoomInLeft',
            },
            'zoomInRight': {
                'animation-name': 'zoomInRight',
            },
            'zoomInUp': {
                'animation-name': 'zoomInUp',
            },
            'bounceIn': {
                'animation-name': 'bounceIn',
                'animation-duration': `${bounceInSpeed}ms`,
            },
            'bounceInDown': {
                'animation-name': 'bounceInDown',
            },
            'bounceInLeft': {
                'animation-name': 'bounceInLeft',
            },
            'bounceInRight': {
                'animation-name': 'bounceInRight',
            },
            'bounceInUp': {
                'animation-name': 'bounceInUp',
            },
            'bounceOut': {
                'animation-name': 'bounceOut',
                'animation-duration': `${bounceOutSpeed}ms`,
            },
            'bounceOutDown': {
                'animation-name': 'bounceOutDown',
            },
            'bounceOutLeft': {
                'animation-name': 'bounceOutLeft',
            },
            'bounceOutRight': {
                'animation-name': 'bounceOutRight',
            },
            'bounceOutUp': {
                'animation-name': 'bounceOutUp',
            },
            'zoomOut': {
                'animation-name': 'zoomOut',
            },
            'zoomOutDown': {
                'animation-name': 'zoomOutDown',
            },
            'zoomOutLeft': {
                'animation-name': 'zoomOutLeft',
            },
            'zoomOutRight': {
                'animation-name': 'zoomOutRight',
            },
            'zoomOutUp': {
                'animation-name': 'zoomOutUp',
            },
            'slideInDown': {
                'animation-name': 'slideInDown'
            },
            'slideInLeft': {
                'animation-name': 'slideInLeft'
            },
            'slideInRight': {
                'animation-name': 'slideInRight'
            },
            'slideInUp': {
                'animation-name': 'slideInUp'
            },
            'slideOutDown': {
                'animation-name': 'slideOutDown'
            },
            'slideOutLeft': {
                'animation-name': 'slideOutLeft'
            },
            'slideOutRight': {
                'animation-name': 'slideOutRight'
            },
            'slideOutUp': {
                'animation-name': 'slideOutUp'
            },
            'fadeIn': {
                'animation-name': 'fadeIn'
            },
            'fadeInDown': {
                'animation-name': 'fadeInDown'
            },
            'fadeInDownBig': {
                'animation-name': 'fadeInDownBig'
            },
            'fadeInLeft': {
                'animation-name': 'fadeInLeft'
            },
            'fadeInLeftBig': {
                'animation-name': 'fadeInLeftBig'
            },
            'fadeInRight': {
                'animation-name': 'fadeInRight'
            },
            'fadeInRightBig': {
                'animation-name': 'fadeInRightBig'
            },
            'fadeInUp': {
                'animation-name': 'fadeInUp'
            },
            'fadeInUpBig': {
                'animation-name': 'fadeInUpBig'
            },
            'fadeInTopLeft': {
                'animation-name': 'fadeInTopLeft'
            },
            'fadeInTopRight': {
                'animation-name': 'fadeInTopRight'
            },
            'fadeInBottomLeft': {
                'animation-name': 'fadeInBottomLeft'
            },
            'fadeInBottomRight': {
                'animation-name': 'fadeInBottomRight'
            },
            'fadeOut': {
                'animation-name': 'fadeOut'
            },
            'fadeOutDown': {
                'animation-name': 'fadeOutDown'
            },
            'fadeOutDownBig': {
                'animation-name': 'fadeOutDownBig'
            },
            'fadeOutLeft': {
                'animation-name': 'fadeOutLeft'
            },
            'fadeOutLeftBig': {
                'animation-name': 'fadeOutLeftBig'
            },
            'fadeOutRight': {
                'animation-name': 'fadeOutRight'
            },
            'fadeOutRightBig': {
                'animation-name': 'fadeOutRightBig'
            },
            'fadeOutUp': {
                'animation-name': 'fadeOutUp'
            },
            'fadeOutUpBig': {
                'animation-name': 'fadeOutUpBig'
            },
            'backInUp': {
                'animation-name': 'backInUp'
            },
            'backInDown': {
                'animation-name': 'backInDown'
            },
            'backInLeft': {
                'animation-name': 'backInLeft'
            },
            'backInRight': {
                'animation-name': 'backInRight'
            },
            'backOutUp': {
                'animation-name': 'backOutUp'
            },
            'backOutDown': {
                'animation-name': 'backOutDown'
            },
            'backOutLeft': {
                'animation-name': 'backOutLeft'
            },
            'backOutRight': {
                'animation-name': 'backOutRight'
            },
        }

        let utilities = {};
        let keyFrames = {};

        if (classes && classes.length > 0) {
            classes.forEach((el) => {
                utilities[`${el}`] = fallbackUtilities[`${el}`];
                keyFrames[`@keyframes ${el}`] = fallbackKeyframes[`@keyframes ${el}`];
            });
        } else {
            utilities = fallbackUtilities;
            keyFrames = fallbackKeyframes;
        }

        addUtilities(keyFrames)

        utilities = {...utilities, ...fallbackAnimateUtility};

        addUtilities(
            [
                Object.entries(utilities).map(([key, value]) => {
                    return {
                        [`.${e(`${animatedClassName}${key}`)}`]: value,
                    }
                })
            ]
        )
    },
    {
        experimental: {
            matchVariant: true
        },
        theme: {
            animatedSettings: {
                animatedSpeed: 1000,
                heartBeatSpeed: 500,
                hingeSpeed: 2000,
                bounceInSpeed: 750,
                bounceOutSpeed: 750,
                animationDelaySpeed: 500,
                animatedClassName: defaultAnimatedClassName,
                classes: ['infinite', 'bounce', 'heartBeat']
            }
        },
    }
)

module.exports = animateCssTailwindCss;
